auslib/db.py

```diff
@@ -796,22 +797,25 @@ def getRulesMatchingQuery(self, updateQuery, fallbackChannel, transaction=None):
                 self.log.debug(r)
         return matchingRules
 
-    def getRuleById(self, rule_id, transaction=None):
-        """ Returns the unique rule that matches the give rule_id """
-        rules = self.select(where=[self.rule_id == rule_id], transaction=transaction)
+    def getRule(self, id_or_alias, transaction=None):
+        """ Returns the unique rule that matches the give rule_id or alias."""
+        rules = self.select(
+            where=[(self.alias == id_or_alias) | (self.rule_id == id_or_alias)],
```

Comment:
would it be better to figure out if id_or_alias is an id or alias first? IIRC, you can only refer to a single index per table per query, so this could be slow unless we add a new index for both columns, or figure out which column we want to query first.